code_docs/Organization.md

```diff
@@ -11,6 +11,37 @@
 -  Some new menu-items - especially in `community firmware` - don't have descriptive names but are listed as "alternate [x] behavior", which puts the load on the user's memory to remember what they do. It would be helpful to have all menu names be descriptive
 -  At the same time, many menu items are longer than the display allows. This is probably not possible to avoid altogether, but navigating a menu with items that fit the display is significantly faster. It could pay off to rework menu items' naming to be both descriptive and short. For example: `stutter quantize` conveys as much information as `stutter rate quantize` while still fitting on the display. *(Note: some longer names from the community features menu will get shorter automatically when they are implemented in their dedicated spot in the regular menu, f.e. `grid view loop layer pads` will become `loop layer pads` when placed in the grid view menu)*
+#### Modes vs Views vs Layouts
+If there is a learning curve to the Deluge, I feel the biggest hurdle is the lack of distinction between the scope and significance of different views. Currently, everything from the arranger, to the grid, to sequencing, to audio recording, to loading files is called a **view**. Having no structure to place all these fundamentally different features into makes it difficult to build up a good understanding what is happening.
+
+Having a more descriptive, hierarchical naming convention for different kinds of features makes it easier for beginners to feel at ease with the interface, *and* makes it easier for developers to add features in a way that keeps the interface manageable. This could be part of refactoring ui & views, if we decide to do so. A proposal:
+
+1. **Modes** describe mutually exclusive fundamental ways in which the device behaves. Currenly this would only apply to **Song Mode** and **Arranger Mode**. At any time, either one or the other is active - never both. This allows us to rename the Song Mode layouts to **Rows View** and **Grid View**. *This could be implemented without changing anything else.*
+2. **Views** exclusively describe ways of **viewing** and **interacting** with a **mode**. Currently this would only apply to **Rows View**, **Grid View** and **Performance View**. *(One could imagine a grid view for arranger mode, where the arrangement plays top to bottom, which would make all views compatible with both modes.)*
+3. **Editors:** clip and automation view operate fundamentally different from the more similar aforementioned views. To distinguish them from those views, they are labeled **Clip Editor**, **Clip Automation Editor** and **Arranger Automation Editor**.
```

Comment:
I like where this is going. We're definitely on the same page regarding Modes (Song vs Arranger) and the Views to view and interact with the modes.

Regarding the editor concept. I think this term makes sense to me. You are basically editing a component of the project.

Regarding Automation View tho, it's a bit complex of a beast in the sense, it has an "Overview" component which basically acts as a "Report" / "Launchpad" so to speak which can bring you into the Editor.

Also the automation view currently has two UI's for editing parameter automation as well as note velocity. Those are definitely editor's and I agree with the term editor there:
- Parameter Automation Editor
- Note Velocity Editor

We'll also eventually have other editors too:
- Note Probability Editor
- Note Iteration Editor
- Note Expression Editor

So I'm not entirely sure how to present that to a user.

The way I see it, it's a UI for interacting with and editing various parameters, whether they be modulation or note parameters. 

I think the term Automation is familiar to users when it comes to editing parameters like this.

When it comes to editing clips, we basically have 3 editing layouts:

- audio clip
- melodic instrument clip (synth, midi, cv)
- kit clip

Reply:
Re: automation view - see following commit

Re: editing clips - from a programmer's perspective, layouts would be a good label here. From the user's perspective, I would just call it "Editing audio clips in the clip editor." As they don't have any say over what layout they get, we might as well not label it